exporter/collector/googlemanagedprometheus/monitoredresource.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlemanagedprometheus
+
+import (
+	"go.opentelemetry.io/collector/model/pdata"
+	semconv "go.opentelemetry.io/collector/model/semconv/v1.8.0"
+	monitoredrespb "google.golang.org/genproto/googleapis/api/monitoredres"
+)
+
+func MapToPrometheusTarget(res pdata.Resource) *monitoredrespb.MonitoredResource {
+	attrs := res.Attributes()
+	// Prepend namespace if it exists to match what is specified in
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#resource-attributes-1
+	job := getStringOrEmpty(attrs, semconv.AttributeServiceName)
+	serviceNamespace := getStringOrEmpty(attrs, semconv.AttributeServiceNamespace)
+	if serviceNamespace != "" {
+		job = serviceNamespace + "/" + job
+	}
+	return &monitoredrespb.MonitoredResource{
+		Type: "prometheus_target",
+		Labels: map[string]string{
+			"location":  getStringOrEmpty(attrs, semconv.AttributeCloudAvailabilityZone),
+			"cluster":   getStringOrEmpty(attrs, semconv.AttributeK8SClusterName),
+			"namespace": getStringOrEmpty(attrs, semconv.AttributeK8SNamespaceName),
```

Comment:
For Managed GMP, the namespace is renamed from `__meta_kubernetes_namespace`. 
(https://github.com/GoogleCloudPlatform/prometheus-engine/blob/main/pkg/operator/apis/monitoring/v1/types.go#L381).

For unmanaged GMP, IIRC we rely users to set the `namespace` label via global labels or relabeling rules in their Prometheus configuration.

Does that match the behavior here?

Reply:
Soon ([open-telemetry/opentelemetry-collector-contrib#9416](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/9416)), the prometheus receiver will automatically grab `__meta_kuberentes_namespace` from scrape SD labels without requiring any relabel rules.

Reply:
Thanks. If users do configure a relabel rule to set the `namespace` label, which takes precedence?

Reply:
Relabel rules which set `namespace` won't have any impact here.  It would add a label named "namespace" to the metric.  To set the namespace attribute (e.g. outside of k8s, or for a static target), someone would have to use the resource processor:
```yaml
resource:
  attributes:
  - key: k8s.namespace.name
    value: my-custom-namespace
    action: upsert
```